extensions/sql-migration/src/dialog/validationResults/valideIRDialog.ts

```diff
@@ -0,0 +1,658 @@
+			await this._initTestIrResults(results);
+		}
+	}
+
+	private async _validate(): Promise<void> {
+		this._canceled = false;
+		await this._initializeResults();
+
+		if (this._model._targetType === MigrationTargetType.SQLDB) {
+			await this._validateSqlDbMigration();
+		} else {
+			await this._validateDatabaseMigration();
+		}
+
+		this._saveResults();
+	}
+
+	private async _validateDatabaseMigration(): Promise<void> {
+		const sqlConnections = await azdata.connection.getConnections();
+		const currentConnection = sqlConnections.find(
+			value => value.connectionId === this._model.sourceConnectionId);
+		const sourceServerName = currentConnection?.serverName!;
+		const trustServerCertificate = currentConnection?.options?.trustServerCertificate === true;
+		const databaseCount = this._model._databasesForMigration.length;
+		const sourceDatabaseName = this._model._databasesForMigration[0];
+		const networkShare = this._model._databaseBackup.networkShares[0];
+
+		try {
+			await this._updateValidateIrResults(0, ValidateIrState.Running);
+			// valdiate IR is online
+			const response = await validateIrDatabaseMigrationSettings(
+				this._model,
+				sourceServerName,
+				trustServerCertificate,
+				sourceDatabaseName,
+				networkShare,
+				true,
+				false,
+				false,
+				false);
+			if (response?.errors?.length > 0) {
+				const errors = response.errors.map(
+					error => constants.VALIDATE_IR_VALIDATION_RESULT_ERROR(
+						sourceDatabaseName,
+						networkShare.networkShareLocation,
+						error));
+				await this._updateValidateIrResults(0, ValidateIrState.Failed, errors);
+			} else {
+				await this._updateValidateIrResults(0, ValidateIrState.Succeeded);
+			}
+		} catch (error) {
+			await this._updateValidateIrResults(0, ValidateIrState.Failed, [constants.VALIDATE_IR_VALIDATION_RESULT_API_ERROR(sourceDatabaseName, error)]);
+		}
+
+		try {
+			await this._updateValidateIrResults(1, ValidateIrState.Running);
+
+			// validate blob container connectivity
+			const response = await validateIrDatabaseMigrationSettings(
+				this._model,
+				sourceServerName,
+				trustServerCertificate,
+				sourceDatabaseName,
+				networkShare,
+				false,
+				false,
+				false,
+				true);
+			if (response?.errors?.length > 0) {
+				const errors = response.errors.map(
+					error => constants.VALIDATE_IR_VALIDATION_RESULT_ERROR(
+						sourceDatabaseName,
+						networkShare.networkShareLocation,
+						error));
+				await this._updateValidateIrResults(1, ValidateIrState.Failed, errors);
+			} else {
+				await this._updateValidateIrResults(1, ValidateIrState.Succeeded);
+			}
+		} catch (error) {
+			await this._updateValidateIrResults(1, ValidateIrState.Failed, [constants.VALIDATE_IR_VALIDATION_RESULT_API_ERROR(sourceDatabaseName, error)]);
+		}
+
+		for (let i = 0; i < databaseCount; i++) {
+			if (this._canceled) {
+				await this._updateValidateIrResults(i + 2, ValidateIrState.Canceled, [`Validation check canceled`])
```

Comment:
I see what you're doing here, but the all the `i + 2`s in every call to `_updateValidateIrResults`  can be a little confusing. Would it maybe make more sense to re-write the `for` loop so that instead of starting at 0 and ending at `databaseCount`, it could start at 2 and end at `databaseCount + 2`, and that way we can use the value of `i` directly?